src/frontend/src/session.rs

```diff
@@ -615,10 +615,47 @@ impl Session for SessionImpl {
             )));
         }
         let stmt = stmts.swap_remove(0);
-        let rsp = infer(self, stmt, sql).map_err(|e| {
-            tracing::error!("failed to handle sql:\n{}:\n{}", sql, e);
-            e
-        })?;
+        let rsp = match stmt {
+            Statement::Query(_) => infer(self, stmt, sql).map_err(|e| {
+                tracing::error!("failed to handle sql:\n{}:\n{}", sql, e);
+                e
+            })?,
+            Statement::ShowObjects(show_object) => match show_object {
+                ShowObject::Columns { table: _ } => {
+                    vec![
+                        PgFieldDescriptor::new("Name".to_owned(), TypeOid::Varchar),
+                        PgFieldDescriptor::new("Type".to_owned(), TypeOid::Varchar),
+                    ]
+                }
```

Comment:
Do you take some ref on our previous code for this part?

Reply:
Yes. This part I refer from the handle(frontend/src/handler/mod.rs/handle) and I extract the related part of PgFieldDescription.

Reply:
Add some comments for where refer from.